app/models/concerns/products/relay_support.rb

```diff
@@ -64,8 +64,13 @@ def check_relay_with_right_type
   # causes too much trouble. Just get rid of the old relay and
   # setup from scratch.
   def destroy_and_init_relay
+    # This instance variable is only present when using the instrument edit form.
+    # We're adding this check to avoid any autosave or association validations
+    # from triggering this callback.
+    return unless @control_mechanism
+
     attrs = relay.try(:attributes) || {}
-    relay.try :destroy
+    relay&.destroy
```

Comment:
Why are we changing the `try` for this other safe navigation operator, does it make a particular difference? Not asking to change it, just wondering why.

Reply:
`try` is kind of a funky method in that it's more of a `object.send(:method_name) if object.respond_to?(:method_name)` than a nil check. It will still respond with a `nil` if the the object is present, but doesn't not have that method. Or at least that's how it worked at some point.

`try` existed in Rails way before the safe navigation operator and this is a really old method (nucore was on ruby 1.8.7 at one point). And  I personally think the safe operator is clearer in that it implies you're doing a nil check.